File: VB/T415077/Form1.vb
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraVerticalGrid.Rows
Imports System

Namespace T415077

    Public Partial Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim file As Data.File = New Data.File() With {.Path = "C:\Home", .Path2 = "C:\Public"}
            propertyGridControl1.SelectedObject = file
        End Sub

        Private Sub edit_ButtonClick(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.ButtonPressedEventArgs)
            openFileDialog1.ShowDialog()
        End Sub

        Private Sub Form1_Shown(ByVal sender As Object, ByVal e As EventArgs)
            Dim edit As RepositoryItemButtonEdit = New RepositoryItemButtonEdit()
            AddHandler edit.ButtonClick, AddressOf edit_ButtonClick
            TryCast(propertyGridControl1.Rows(0), CategoryRow).ChildRows("rowPath2").Properties.RowEdit = edit
        End Sub
    End Class
End Namespace

File: CS/T415077/Form1.cs
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraVerticalGrid.Rows;
using System;

namespace T415077 {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            Data.File file = new Data.File() { Path = "C:\\Home", Path2 = "C:\\Public" };
            this.propertyGridControl1.SelectedObject = file;
        }
        void edit_ButtonClick(object sender, DevExpress.XtraEditors.Controls.ButtonPressedEventArgs e) {
            this.openFileDialog1.ShowDialog();
        }
        private void Form1_Shown(object sender, EventArgs e) {
            RepositoryItemButtonEdit edit = new RepositoryItemButtonEdit();
            edit.ButtonClick += edit_ButtonClick;
            (this.propertyGridControl1.Rows[0] as CategoryRow).ChildRows["rowPath2"].Properties.RowEdit = edit;
        }
    }
}
